lisp/ess-r-mode.el

```diff
@@ -2260,7 +2262,36 @@ state.")
         ess-help-sec-regex ess-help-r-sec-regex
         ess-help-sec-keys-alist ess-help-r-sec-keys-alist ; TODO: Still necessary?
         inferior-ess-help-command inferior-ess-r-help-command
-        inferior-ess-help-filetype nil))
+        inferior-ess-help-filetype nil)
+  (ess-r-help-add-links))
+
+(define-button-type 'ess-r-help-link
+  'follow-link t
+  'action (lambda (_) (ess-r-help-button-action)))
+
+(defun ess-r-help-button-action ()
+  "Display help for button at point."
+  (let ((text (get-text-property (point) 'ess-r-help-link-text)))
+    (ess-display-help-on-object text)))
+
+(defun ess-r-help-add-links ()
+  "Add links to the help buffer."
+  (save-excursion
+    ;; Search for fancy quotes only. If users have
+    ;; options(useFancyQuotes) set to something other than TRUE this
+    ;; probably won't work. If it's FALSE, R outputs ascii ', but
+    ;; searching through the whole buffer takes too long.
+    (while (re-search-forward "‘.+?’" nil t)
```

Comment:
I think we can assume that help topics don't contain spaces. This might help with false positives. 

The non-greedy `+?` doesn't do much here I think.

Reply:
True, I'll add that in. 

We need the non greedy regex because otherwise if a line contains two matches we match from the beginning of the first to the end of the second. Given `foo 'bar' foo 'baz'` we'd buttonize `'bar' foo 'baz'` as one link.